### DISCUSSION TITLE
Help getting Elecrow 7 inch HMI ESP32 display working?

### DISCUSSION THREAD

**Discussion by JeffreyRon** [moononournation/Arduino_GFX#367] [Q&A]
I have the Elecrow 5 inch HMI ESP32 display and it works great.  It's listed in the Dev-Device-Declaration under 'Elecrow Wizee-ESP32 WZ8048C050. I liked it so much I bought the 7 inch (WZ8048C070), but after hours of trying, I can't get it to work. Just get a white screen. It did come with their "factory" LVGL demo program and it was working, so I know the unit works. I also wrote small sketches to send info to serial and that works. So it comes down to the video information. It is supposed to be very similar to the 5 inch, but the 5 inch uses ILI6122 & ILI5960 driver and the 7 inch uses EK9716BD3 & EK73002ACGB driver. I've even tried reloading their factory demo program without luck. I had no problem reloading the 5 inch factory demo program on the 5 inch. 

I've tried using the information from the factory demo program to recreate the Arduino_ESP32RGBPanel info, but still no luck. I wish they would provide a simple Hello World sketch instead of that bloated, overly edited factory demo.

ESP32 display-7.0 Inch HMI Display 800x480 RGB TFT LCD
https://www.elecrow.com/esp32-display-7-inch-hmi-display-rgb-tft-lcd-touch-screen-support-lvgl.html

Below is the 5 inch factory demo parameters and the Arduino_GFX dev list parameters. I can get both to work. The factory demo parameters require an older version of Arduino_GFX that comes with their libraries download. So I don't use it, I use the latest GFX_For_Arduino version from the dev list.

From the Elecrow 5 inch factory program (working - with older version of Arduino_GFX)

Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  GFX_NOT_DEFINED /* CS */, GFX_NOT_DEFINED /* SCK */, GFX_NOT_DEFINED /* SDA */,
  40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
  45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
  5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
  8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */);

Arduino_RPi_DPI_RGBPanel *lcd = new Arduino_RPi_DPI_RGBPanel(
  bus,
  800 /* width */, 0 /* hsync_polarity */, 210 /* hsync_front_porch */, 4 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
  480 /* height */, 0 /* vsync_polarity */, 22 /* vsync_front_porch */, 4 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 16000000 /* prefer_speed */, true /* auto_flush */);

------------------------------------------------------------------------------------------------------------------------
5 inch from Arduino_GFX Dev list (working with latest version of GFX_For_Arduino)

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
  40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
  45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
  5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
  8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
  0 /* hsync_polarity */, 8 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 32 /* hsync_back_porch */,
  0 /* vsync_polarity */, 8 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 8 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 16000000 /* prefer_speed */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(800 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */);

------------------------------------------------------------------------------------------------------------------------
Now here are the 7 inch. Neither work!

From the Elecrow 7 inch factory program  NOT working on reload attempts. Demo program was loaded and working when I received the display.

Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  GFX_NOT_DEFINED /* CS */, GFX_NOT_DEFINED /* SCK */, GFX_NOT_DEFINED /* SDA */,
  41 /* DE */, 40 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
  14 /* R0 */, 21 /* R1 */, 47 /* R2 */, 48 /* R3 */, 45 /* R4 */,
  9 /* G0 */, 46 /* G1 */, 3 /* G2 */, 8 /* G3 */, 16 /* G4 */, 1 /* G5 */,
  15 /* B0 */, 7 /* B1 */, 6 /* B2 */, 5 /* B3 */, 4 /* B4 */);

Arduino_RPi_DPI_RGBPanel *lcd = new Arduino_RPi_DPI_RGBPanel(
  bus,
  800 /* width */, 0 /* hsync_polarity */, 210 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 46 /* hsync_back_porch */,
  480 /* height */, 0 /* vsync_polarity */, 22 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 23 /* vsync_back_porch */,
  0 /* pclk_active_neg */, 16000000 /* prefer_speed */, true /* auto_flush */);

----------------------------------------------------------------------------------------------------------------------
7 inch attempt to recreate from the above information.

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
  41 /* DE */, 40 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
  14 /* R0 */, 21 /* R1 */, 47 /* R2 */, 48 /* R3 */, 45 /* R4 */,
  9 /* G0 */, 46 /* G1 */, 3 /* G2 */, 8 /* G3 */, 16 /* G4 */, 1 /* G5 */,
  15 /* B0 */, 7 /* B1 */, 6 /* B2 */, 5 /* B3 */, 4 /* B4 */,
  0 /* hsync_polarity */, 210 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 46 /* hsync_back_porch */,
  0 /* vsync_polarity */, 22 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 23 /* vsync_back_porch */,
  0 /* pclk_active_neg */, 16000000 /* prefer_speed */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(800 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */);

**Comment by moononournation** [comment]
I think you first need to know why factory image not work

**Comment by DaveDuncanson** [comment]
Hi Jeff
I am having exactly the same issue, with the Elecrow 7" display.  I'm using the PlatformIO environment and attempting to port openHASP to the Elecrow 7" inch and not matter what I have tried, including earlier versions of the ArdunioGFX library (1.30 & 1.2.8), all I get on the display is a white screen. I also spent days trying to get the example code from the Elecrow Wiki working, without any success, using both Ardunio and PlatformIO as per the Wiki instructions. 

The settings Im using for the TFT are : 
 ; Panel Settings
    -D TFT_WIDTH=800
    -D TFT_HEIGHT=480

    -D TFT_HSYNC_POLARITY=1
    -D TFT_HSYNC_FRONT_PORCH=40   ; Maximum HSYNC Front Porch
    -D TFT_HSYNC_PULSE_WIDTH=48   ; Typical HSYNC Pulse Width
    -D TFT_HSYNC_BACK_PORCH=40    ; Typical HSYNC Back Porch
    -D TFT_VSYNC_POLARITY=1
    -D TFT_VSYNC_FRONT_PORCH=13   ; Maximum VSYNC Front Porch
    -D TFT_VSYNC_PULSE_WIDTH=1    ; Typical VSYNC Pulse Width
    -D TFT_VSYNC_BACK_PORCH=31    ; Typical VSYNC Back Porch
    -D TFT_PCLK_ACTIVE_NEG=1
    -D TFT_PREFER_SPEED=16000000   ; Typical DCLK Frequency
    -D TFT_AUTO_FLUSH=1

 ; Bus Settings
    -D TFT_DE=41
    -D TFT_VSYNC=40
    -D TFT_HSYNC=39
    -D TFT_PCLK=42 
    -D TFT_R0=14 
    -D TFT_R1=21
    -D TFT_R2=47
    -D TFT_R3=48
    -D TFT_R4=45
    -D TFT_G0=9
    -D TFT_G1=46
    -D TFT_G2=3
    -D TFT_G3=8
    -D TFT_G4=16
    -D TFT_G5=1
    -D TFT_B0=15
    -D TFT_B1=7
    -D TFT_B2=6
    -D TFT_B3=5
    -D TFT_B4=4

The touch screen is working fine with : 

    ; Touch Settings
    -D TOUCH_WIDTH=800
    -D TOUCH_HEIGHT=480
    -D TOUCH_DRIVER=0x911
    -D TOUCH_SCL=20
    -D TOUCH_SDA=19
    -D TOUCH_IRQ=-1
    -D TOUCH_RST=-1
    -D I2C_TOUCH_ADDRESS=0x5D 
    -D I2C_TOUCH_FREQUENCY=400000

GFX Initialisation: 

    Arduino_RGBPanel_Mod* bus = new Arduino_RGBPanel_Mod(
        TFT_DE, TFT_VSYNC, TFT_HSYNC, TFT_PCLK, TFT_R0, TFT_R1, TFT_R2, TFT_R3, TFT_R4, TFT_G0, TFT_G1, TFT_G2, TFT_G3,
        TFT_G4, TFT_G5, TFT_B0, TFT_B1, TFT_B2, TFT_B3, TFT_B4, TFT_HSYNC_POLARITY, TFT_HSYNC_FRONT_PORCH,
        TFT_HSYNC_PULSE_WIDTH, TFT_HSYNC_BACK_PORCH, TFT_VSYNC_POLARITY, TFT_VSYNC_FRONT_PORCH, TFT_VSYNC_PULSE_WIDTH,
        TFT_VSYNC_BACK_PORCH, TFT_PCLK_ACTIVE_NEG, TFT_PREFER_SPEED);

    Arduino_RGB_Display_Mod* gfx = new Arduino_RGB_Display_Mod(TFT_WIDTH, TFT_HEIGHT, bus);
.
One clue, is that after initialising the GFX, if I call    
gfx->fillScreen(RED);

I get an exception error. 

Rgs
Dave

**Reply by moononournation** [reply]
if you want using Arduino_GFX, please stickly using Arduino IDE and test with PDQgraphicstest example first.

**Reply by DaveDuncanson** [reply]
Hi 
Okay, will do, I will switch to to the Arduino client tomorrow and will run  the above example t(PDQgraphicstest) to see what happens. 
Rgs
Dave

**Reply by DaveDuncanson** [reply]
Hi 
Just built and uploaded via the PDQgraphicstest example via the Ardunio Client with the following results
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x0 (DOWNLOAD(USB/UART0))
waiting for download
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x44c
load:0x403c9700,len:0xbd8
load:0x403cc700,len:0x2a80
entry 0x403c98d0
Arduino_GFX library Test!
Benchmark	micro-secs
Screen fill	164464
Text	20103
Pixels	923157
Lines	524151
Horiz/Vert Lines	14523
Rectangles (filled)	341785
Rectangles (outline)	9677
Triangles (filled)	125025
Triangles (outline)	29820
Circles (filled)	57688
Circles (outline)	50326
Arcs (filled)	36336
Arcs (outline)	69228
Rounded rects (filled)	348885
Rounded rects (outline)	25388
Done!
Benchmark	micro-secs
Screen fill	164464
Text	20031
Pixels	923157
Lines	524152
Horiz/Vert Lines	14523
Rectangles (filled)	341785
Rectangles (outline)	9677
Triangles (filled)	125003
Triangles (outline)	29820
Circles (filled)	57688
Circles (outline)	50325
Arcs (filled)	36284
Arcs (outline)	69227
Rounded rects (filled)	348884
Rounded rects (outline)	25378
Done!
Benchmark	micro-secs
Screen fill	164464
Text	20031
Pixels	923157
Lines	524151
Horiz/Vert Lines	14523
Rectangles (filled)	341785
Rectangles (outline)	9658
Triangles (filled)	125003
Triangles (outline)	29820
Circles (filled)	57689
Circles (outline)	50325
Arcs (filled)	36301
Arcs (outline)	69227
Rounded rects (filled)	348885
Rounded rects (outline)	25378
Done!

**Reply by JeffreyRon** [reply]
Yes, I've tried everything. As mentioned, I've written small sketches that send data to the serial port, and that works. Plus my hello world sketch which is super simple almost works, by that I mean I can barely make out the 'He' it the top left corner for just a second, then it just goes all white. The 5 inch works so great! If only this 7 inch worked as well. I'm sure it's just the video settings. My only hope was to try and reconstruct those setting from information in the factory demo program, but obviously no luck. Again, it did come with the factory demo running, but I can't get it to reload. I remember reading an Amazon review where another guy said he couldn't get the factory demo program to reload either.

**Reply by DaveDuncanson** [reply]
Same, the functionality of the core app (openHASP) which includes a number of services (MQTT, HTTP, WiFI, etc) all work well, just nothing on the screen and when I do the same build for other TFTs, all works well including the display. I also agree that I suspect the issue is related to the video parameters, so I have also confirmed all of the pin assignments from the example code against the schematic and board layouts and they all seem to be correct.  I'm now thinking about doing the same with the TFT datasheet against the TFT panel parameters (ie HSYNC, VSYNC,etc). I am also planning on posting something on  the OEMs (Elecrow) support pages, asking for an updated version of the example code for the 7inch that actually works :)

**Comment by DaveDuncanson** [comment]
Just stumbled across this in the OEM support forum, I'm going to give this a go next as I can easily build with either library, 
https://forum.elecrow.com/discussion/473/lovyangfx-settings-for-7-display

**Comment by DaveDuncanson** [comment]
Hi Ron
Switching to the LovyanGFX library (v1.1.8) with the parameters from the above link has WORKED. I now have the expected graphic controls on the screen. While I was converting the panel parameters across, I did notice some changes to the HSYNC, VSYNC polarity  settings and the speed, from what I had from the example code, which _may_ explain the original issue. If I get some time this week, I will try going back to ArduinoGFX with the updated parameters from below, to see if that resolves the original issue.  

For now, I'm just happy that I have something on the screen.

    -D TFT_HSYNC_POLARITY=0
    -D TFT_HSYNC_FRONT_PORCH=40   ; Maximum HSYNC Front Porch
    -D TFT_HSYNC_PULSE_WIDTH=48   ; Typical HSYNC Pulse Width
    -D TFT_HSYNC_BACK_PORCH=40    ; Typical HSYNC Back Porch
    -D TFT_VSYNC_POLARITY=0
    -D TFT_VSYNC_FRONT_PORCH=1   ; Maximum VSYNC Front Porch
    -D TFT_VSYNC_PULSE_WIDTH=31    ; Typical VSYNC Pulse Width
    -D TFT_VSYNC_BACK_PORCH=13    ; Typical VSYNC Back Porch
    -D TFT_PCLK_ACTIVE_NEG=1
    -D TFT_PREFER_SPEED=12000000   ; Typical DCLK Frequency
    -D TFT_AUTO_FLUSH=1
    -D TFT_PCLK_ACTIVE_NEG=1

**Reply by moononournation** [reply]
Please note Arduino_GFX does not require any -D compile parameters. And all TFT_* parameters not related to Arduino_GFX, so it have no any effect you setting it.

**Reply by DaveDuncanson** [reply]
Thanks for  highlighting  and yes the -D TFT_* is from the code base from the core app that I'm adding support for the Elecrow 7inch  and all these will need need converting back into the respective arguments for both Ardunio_GFX and or LovyanGFX libraries.

**Reply by JeffreyRon** [reply]
Wow, guys. Glad to hear there is some progress. I was worried this 7 inch was just going to be a paperweight. Can't wait to see it working.

**Comment by DaveDuncanson** [comment]
Hi Ron

I just updated the Elecrow example code for the 7inch, using Ardunio,  with the same parameters as used above (LGFX) in the equivalent call to the ArdunioGFX library and I now have the original example application working again on the device.

So as we suspected the issue is with the Panel/Video parameters and not the GFX libraries. So the calls that need to be updated are  below and I'm using V1.2.8 of the ArdunioGFx library, not the latest version (as per the instructions from Elecrow's Wiki)

Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  GFX_NOT_DEFINED /* CS */, GFX_NOT_DEFINED /* SCK */, GFX_NOT_DEFINED /* SDA */,
  41 /* DE */, 40 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
  14 /* R0 */, 21 /* R1 */, 47 /* R2 */, 48 /* R3 */, 45 /* R4 */,
  9 /* G0 */, 46 /* G1 */, 3 /* G2 */, 8 /* G3 */, 16 /* G4 */, 1 /* G5 */,
  15 /* B0 */, 7 /* B1 */, 6 /* B2 */, 5 /* B3 */, 4 /* B4 */
);

Arduino_RPi_DPI_RGBPanel *lcd = new Arduino_RPi_DPI_RGBPanel(
  bus,
  800 /* width */, 0 /* hsync_polarity */, 40 /* hsync_front_porch */, 48 /* hsync_pulse_width */, 40 /* hsync_back_porch */,
  480 /* height */, 0 /* vsync_polarity */, 1 /* vsync_front_porch */, 31 /* vsync_pulse_width */, 13 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 16000000 /* prefer_speed */, true /* auto_flush */);

**Comment by DaveDuncanson** [comment]
Just for completeness, the following calls work with the latest version of ArdunioGFX library (1.3.8) 

Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
    41 /* DE */, 40 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
    14 /* R0 */, 21 /* R1 */, 47 /* R2 */, 48 /* R3 */, 45/* R4 */,
    9 /* G0 */, 46 /* G1 */, 3 /* G2 */, 8 /* G3 */, 16 /* G4 */, 1 /* G5 */,
    15 /* B0 */, 7 /* B1 */, 6 /* B2 */, 5 /* B3 */, 4 /* B4 */,
    0 /* hsync_polarity */, 40 /* hsync_front_porch */, 48 /* hsync_pulse_width */, 40 /* hsync_back_porch */,
    0 /* vsync_polarity */, 1 /* vsync_front_porch */, 31 /* vsync_pulse_width */, 13 /* vsync_back_porch */);

    Arduino_RGB_Display *lcd = new Arduino_RGB_Display(
    800 /* width */, 480 /* height */, bus, 0 /* rotation */, true /* auto_flush */);

**Reply by JeffreyRon** [reply]
Such awesome work Dave! You are the man. This is all I needed. (BTW I see that github comments strips out the * )

Here is my hello world sketch:

// Elecrow HMI 7 Inch - 800x480
#include <U8g2lib.h>  // Required if you are using U8g2 fonts
#include <Arduino_GFX_Library.h>

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
41 /* DE */, 40 /* VSYNC */, 39 /* HSYNC */, 0 /* PCLK */,
14 /* R0 */, 21 /* R1 */, 47 /* R2 */, 48 /* R3 */, 45 /* R4 */,
9 /* G0 */, 46 /* G1 */, 3 /* G2 */, 8 /* G3 */, 16 /* G4 */, 1 /* G5 */,
15 /* B0 */, 7 /* B1 */, 6 /* B2 */, 5 /* B3 */, 4 /* B4 */,
0 /* hsync_polarity */, 40 /* hsync_front_porch */, 48 /* hsync_pulse_width */, 40 /* hsync_back_porch */,
0 /* vsync_polarity */, 1 /* vsync_front_porch */, 31 /* vsync_pulse_width */, 13 /* vsync_back_porch */,
1 /* pclk_active_neg */, 16000000 /* prefer_speed */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(800 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */);

#define GFX_BL 2

void setup() {

  pinMode(GFX_BL, OUTPUT);
  digitalWrite(GFX_BL, HIGH);

  gfx->begin();
  gfx->fillScreen(BLACK);
  gfx->setCursor(10, 30);
  gfx->setFont(u8g2_font_maniac_tr);
  gfx->setTextColor(WHITE);
  gfx->drawRect(0, 0, 800, 480, ORANGE);
  gfx->println("Hello World!");

}

void loop() {

}

**Reply by JeffreyRon** [reply]
Dave thanks again for your work in getting this going. It's much appreciated. I wanted to add that when comparing the 5 inch and the 7 inch, the 5 inch definitely has much darker blacks, where with the 7 inch the background looks a lot more gray. If you look at it at just the right angle, the gray turns kind of black, but most viewing angles, it appears to bright. I wrote a sketch to adjust the brightness up and down and again compared the 5 inch and 7 inch, with the same results.  I've attached a picture that demonstrates this. In the picture both are set at 255 brightness. The whole reason I mention this is because I'm not sure if the video parameters have anything to do with this or if it is just the quality of the screen. Again great work!!!

![elecrow5-7](https://github.com/moononournation/Arduino_GFX/assets/149099810/8f46af9c-9d98-41e3-9627-8643b1956d9e)

**Reply by jancha** [reply]
Trying this on[ elecrow 5"](https://www.elecrow.com/esp32-display-5-inch-hmi-display-rgb-tft-lcd-touch-screen-support-lvgl.html) display. Getting the following error:

22:04:26.812 -> E (127) lcd_panel.rgb: esp_lcd_new_rgb_panel(150): no mem for frame buffer
22:04:26.812 -> ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x4037a900
22:04:26.843 -> file: "/Users/jv/Documents/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_ESP32RGBPanel.cpp" line 118
22:04:26.843 -> func: uint16_t* Arduino_ESP32RGBPanel::getFrameBuffer(int16_t, int16_t)
22:04:26.843 -> expression: esp_lcd_new_rgb_panel(_panel_config, &_panel_handle)

(using above mentioned rgbpanel setup).

any ideas on how to address?

thanks,
J

**Reply by JeffreyRon** [reply]
Not sure if you are copying the information from above, but the github comments section removes the * so everything gets messed up. Try this super simple Hello World sketch for the 5 inch.

[Hello_World.zip](https://github.com/moononournation/Arduino_GFX/files/13261370/Hello_World.zip)

**Reply by jancha** [reply]
Hello @JeffreyRon,

Trie hello world and happy to report that it finally works - however had to finetune Arduino IDE settings:

<img width="635" alt="image" src="https://github.com/moononournation/Arduino_GFX/assets/540255/69a1cb40-7294-4de6-a7c8-9a05de294656">

Apparently PSRAM was not enabled, causing the issues.

So happy to report that the sample code with the config above finally works :) What a relief ;)

Thanks,
J.

**Comment by DaveDuncanson** [comment]
Thanks and happy that I could assist, noting that I also was very keen to get the 7 inch version working to replace a very expensive 4.3" touch screen in a project that I'm updating at the moment ( http://smartcoop.tech ). 

Interesting display differences between the 5 and 7 inch versions and I'm no expert on TFT, but from going through the OEMs Wiki several times, I did notice that the 7 inch is using a different Display Driver (EK9716BD3/EK73002ACGB), than the other models, so that may be the cause, but a the same time its quite a large difference in the display characteristics above. 

Just a suggestion, (I don't have a 5 inch version), may be worth drawing on both displays a series of shapes with different primary and secondary colours, to see if the difference is related to a specific colour and therefore I/O pin assignments. Also may be worth posting to the Elecrow support wifi the above photos and example code and ask if they can explain the differences. When I get a chance, I will load your example code on my 7inch screen and then upload some photos to see if we are getting the same results.

**Reply by JeffreyRon** [reply]
Thanks Dave for continuing to look at this. I'm super excited just to have it working, thanks to you. I've attached a zip file with the actual sketch you see running in that picture. The number scrolling is the number of times it goes through the loop per second. The green and red circles are touch areas for turning the brightness up and down with the resulting brightness setting being displayed also. There is also a cool looking DIGITS font commented out if you want to see what that looks like. 
[Display_List_SpeedTest_Touch.zip](https://github.com/moononournation/Arduino_GFX/files/13226314/Display_List_SpeedTest_Touch.zip)

**Reply by JeffreyRon** [reply]
Dave I've had some success with your great investigative work. But I'm having intermittent problems, so I think something needs tweaked.  

What I'm noticing is when I hit reset button, sometimes it take up to about 5 seconds of blank screen before my actual screen appears. Then when my sketch appears it looks great. but then about 30 seconds or a minute later the screen goes blank. It may kind of flicker a bit, then after about 5-10 seconds my sketch reappears. My sketch has a millis counter on it so I know when the screen goes blank, the sketch is still running, it's just the screen going blank. Let me know if you've experienced anything like that or if you have any ideas. I've attached my "Hello World" sketch which uses the parameters you found in the lovyan lib. See if it will load and how the screen reacts. It's about as minimum size as I could get it and still display text and graphics. This hello world sketch doesn't have much going on so the screen may not blink off. I'll also post another sketch that shows off the graphics a little more using canvas from Adafruit_GFX. You might find it cool. Unfortunately since there is a lot of action on the screen, it does make the screen go blank more often. The sketch with graphics and canvas has multiple .ino files. Forgive me if you already know this, but you can click on any of the .ino files and the project will open up. Each .ino file will be in a tab. When you compile they are added in alphanumeric order. If you haven't tried this technique you might find it interesting in how you can organize your code and not have to include all the tabs since they are all compiled together a one long sketch.

[Hello_World.zip](https://github.com/moononournation/Arduino_GFX/files/13462765/Hello_World.zip)

[Display_List_Canvas_ZZ.zip](https://github.com/moononournation/Arduino_GFX/files/13462785/Display_List_Canvas_ZZ.zip)

**Reply by DaveDuncanson** [reply]
Hi
No I haven't experienced anything like that and I have been using the same parameters in another build (openHASP) using same screen, for several weeks and I haven't had any issues. I just downloaded your  DisplayListCanvasZZ source, did a build and upload and your right, it took a bit to display anything, then I got green and red arrows moving across the screen (nice effect), but after roughly a minute the screen went blank for say 20 seconds and then got the arrows back. Rinse and repeat. 

I just had a quick look at your code, noting my C is a little rusty (I do most things in Java & Eclipse now days) and couldn't see anything  obvious. 

But I will have a play tomorrow, to see if I can isolate the issue. 

Rgs
Dave

**Reply by DaveDuncanson** [reply]
Hi 
Decided to have a play with the code to see if I could isolate the issue and I now suspect it could be one of the timing parameters as the code seems fine. I also updated to the latest ArduinoGFX library and added a number of serial debug statements to ensure it wasn't something simple, but after running for around a minute, the display goes blank, but if you monitor the serial output, you can see that the code is running fine. 

I'm currently designing a new PCB with an ESP-32-S3 (https://esp32s3.com/nanos3.html), so I will have a look tomorrow at the TFT Driver Datasheet to see if the timing parameters are within spec. 

Attached is the updated source with the extra debug statements. 
[Display_List_Canvas_ZZ.zip](https://github.com/moononournation/Arduino_GFX/files/13463602/Display_List_Canvas_ZZ.zip)

Update
BTW. I just notice a buffer overflow debug, when I switched on all warnings, in these two lines :
  ListContents[(LineyCount + 1)][75] = '\0';     (was 76, which is the buffer length)
&
    ListContents[ListCount][75] = '\0';

But the issue with the blank screen still exists.

**Reply by JeffreyRon** [reply]
Dave, Thanks for taking the time to look at my code. It does seem like it must have something to do with the Arduino_GFX display parameters, because the rest of the C++ code is simple and all my other sketches, which some are super simple, all have this same display problem.  Plus I can run all of these sketches on the Elecrow 5" with no problems. I'll adjust the buffer issue you found. Thanks again!

Jeff

**Reply by DaveDuncanson** [reply]
Hi 
Found the Driver Datasheet (link below - Table 6.5.1) and the timing parameters are all aligned with the typical values for 800x480, so I now doubt that is the issue. Did some googling and found this, which I suspect could be the cause, because it does look like a data bus transfer issue, especially with the intermittent nature of it. So in a nutshell, it looks like the display is running at the top end of its capability when pushing a high number of 800x480 updates and the transfer from the frame buffer located in PSRAM is getting in the way. 

The also would explain why I haven't seen the behaviour on the other build I have been using, because it is only putting a few buttons and other controls on the screen and updates are via MQTT, so slow in the scheme of things.

Without putting it on a data bus analyser (I don't have one or access to one), I doubt we will know for sure.  
My take away is that this display is relatively cheap in the scheme of things and will be okay for basic GUI functions, but if pushed to its limits, H/W issues will be exposed.  

https://esp32.com/viewtopic.php?f=12&t=26793

https://forum.lvgl.io/t/display-glitching-while-connecting-to-wifi-esp32s3-8048s043/12075/32?page=2

https://www.phoenixdisplay.com/wp-content/uploads/2015/07/EK9716BD_DS_rev1.0_20131121.pdf

**Reply by JeffreyRon** [reply]
Dave thanks again for your time and efforts. I found this guy having a similar problem. You may have already seen this post. I've tried installing esp32 v2.0.3 but I get compile errors. I'm still working on it just to see if it's possible to find some combination that may make the display work better.

Problems with LCD on ESP32-S3-N4R8 with framework > v2.0.3
https://github.com/espressif/arduino-esp32/issues/8468

**Reply by JeffreyRon** [reply]
Elecrow has posted information about using the latest version of ESP32 library! But they are suggesting you replace Arduino_GFX with LovyanGFX  when using the 7 inch. They say to continue using Arduino_GFX for the 5 inch.

https://forum.elecrow.com/discussion/585/esp32-hmi-demo-code-updated-compatible-with-new-version-of-esp32-package/

"When we first edited the demo code, we based it on the 2.0.3 version of the esp32 package, which resulted in compilation errors for many users who installed the new version of esp32. To solve this problem, we updated the esp32 demo code. The updated content is as follows:"....

**Reply by DaveDuncanson** [reply]
Hi 
Had a quick look at the changes and initially I through the VSYNC parameters had changed (this is edited version), but then realised I had mixed  the parameters for the 5 inch with the 7 inch.   

But they have set the bus speed to 15mbps, which I doubt will make a difference, but I will try it in your test code.

UPDATE:  Nope, got same results, after several minutes the screen went black, but I could see that the code was still running via the USB port.

**Reply by JeffreyRon** [reply]
I tried replacing Arduino_GFX with LovyanGFX like the Elecrow suggests, but I just get a blank screen. I looked at LovyanGFX github posts to see what I might be missing and added things like lcd.init(); etc with my super simple Hello World sketch but still no luck. I've attached the sketch here and you can see it's really simple, but again, blank screen. Not sure what I'm missing.

[Hello_World_LovyanGFX.zip](https://github.com/moononournation/Arduino_GFX/files/13617084/Hello_World_LovyanGFX.zip)

**Reply by moononournation** [reply]
I encourage you keep going on official suggested. And please report your issue in right channel, e.g. official web page. Your issue will persist, if not handled offically.

**Reply by jrodanapolis** [reply]
@JeffreyRon were you able to get this to work? I can get my squareline studio UI file to display, but still cannot get the touch functions to work. If not, I think you may need to try lcd.begin() in place of lcd.init() in your setup, and might need to add some lines for setting the backlight up in a different way. I'm using the configuration that benlye recommends from Aug 4 here:
https://github.com/espressif/arduino-esp32/issues/8468

**Comment by DaveDuncanson** [comment]
Hi - Yep that thread would indicate it may not be a H/W limitation, given changing GFX library seemed to resolve the issue. 
I do like a challenge and given I need to setup a new project for a new build in Visual Studio Code using PlatformIO that I'm planing on starting soon, I will initially drop your test code into it and have a play to see if I can get it to compile with esp32 v2.0.3 and if any specific library combinations resolve the issue. Its just that I'm more familiar with PlatformIO than the Ardunio IDE, both a good IDEs.

I won't get to do this for at least a week, but I will let you know how I go.

**Reply by JeffreyRon** [reply]
I really do appreciate your efforts. Hopefully there is some combination of display settings that will make the display work properly. It's frustrating because the 5 inch version works so great. No problems at all!

I see over on Elecrow forums there are other people having issues and Elecrow continues to say use ESP32 2.0.3 library.

[https://forum.elecrow.com/discussion/578/7-inch-esp32-s3-random-screen-shift](url)

[https://forum.elecrow.com/discussion/561/i-need-help-for-7-inch-esp32s3-display](url)

[https://forum.elecrow.com/discussion/473/lovyangfx-settings-for-7-display](url)

Thanks again,
Jeff

**Comment by jrodanapolis** [comment]
@moononournation 
> I encourage you keep going on official suggested. And please report your issue in right channel, e.g. official web page. Your issue will persist, if not handled offically.

Hi,
I followed the guidelines at this link to change everything to the LovyanGFX library:
https://forum.elecrow.com/discussion/585/esp32-hmi-demo-code-updated-compatible-with-new-version-of-esp32-package/

For some reason, I still can't get the touch to work with my Squareline Studios UI. Can you point me in the right direction on why the touch might not be working? I can see my home screen of the UI on the Elecrow 7" display, but I can't get any touch functionality to work.

I was having the same problem before with the Arduino GFX library... I could get the display to show up, but no touch functionality.

Thank you in advance for your help.

**Comment by ghost** [comment]
hi I have the 5" display, 0705H but can not get working it since days. I am  gone through thr video tutorials. 
But the actuell Arduino IDE is newer than in the video, and I am not so familiar with  arduino ide as I use PlattformIO since years. The donwloadedable zip file for the tutorials do not correspond too. I got the example from Lesson 02 copiled and uploaded, but now the screen is blank.
Can someone a provide me a simple running example code useing lvgl libbrary I con build. Apreceateany help
Regards
Rainer
